desidlas/training/training.py

```diff
@@ -0,0 +1,397 @@
+
+    """
+    # These mean & std dev are used to normalize the score from all 3 loss functions for hyperparam optimize
+    # TODO -- Reconsider these for DESI
+    mean_best_accuracy = 0.02
+    std_best_accuracy = 0.0535
+    mean_best_offset_rmse = 4.0
+    std_best_offset_rmse = 1.56
+    mean_best_coldensity_rmse = 0.33
+    std_best_coldensity_rmse = 0.1
+    normalized_score = (1 - best_accuracy - mean_best_accuracy) / std_best_accuracy + \
+                       (best_offset_rmse - mean_best_offset_rmse) / std_best_offset_rmse + \
+                       (best_coldensity_rmse - mean_best_coldensity_rmse) / std_best_coldensity_rmse
+    
+    return normalized_score
+
+
+
+if __name__ == '__main__':
+    #
+    # Execute batch mode
+    #
+    from Dataset import Dataset
+
+    DEFAULT_TRAINING_ITERS = 100000 #30000
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-s', '--hyperparamsearch', help='Run hyperparam search', required=False, action='store_true', default=False)
+    parser.add_argument('-o', '--output_file', help='Hyperparam csv result file location', required=False, default='batch_results.csv')
+    parser.add_argument('-i', '--iterations', help='Number of training iterations', required=False, default=DEFAULT_TRAINING_ITERS)
+    parser.add_argument('-l', '--loadmodel', help='Specify a model name to load', required=False, default=None)
+    parser.add_argument('-c', '--checkpoint_file', help='Name of the checkpoint file to save (without file extension)', required=False, default="train_1016snr5smooth/current") #../models/training/current
+    parser.add_argument('-r', '--train_dataset_filename', help='File name of the training dataset without extension', required=False, default="/home/bwang/data/smoothtrain/16804-snr5dataset.npy")
+    parser.add_argument('-e', '--test_dataset_filename', help='File name of the testing dataset without extension', required=False, default="/home/bwang/data/smoothtrain/2181-snr5dataset.npy")
+    parser.add_argument('-t', '--INPUT_SIZE', help='set the input data size', required=False, default=600)
+    parser.add_argument('-m', '--matrix_size', help='set the matrix size when using smooth', required=False, default=4)
+    args = vars(parser.parse_args())
+
+    RUN_SINGLE_ITERATION = not args['hyperparamsearch']
+    checkpoint_filename = args['checkpoint_file'] if RUN_SINGLE_ITERATION else None
+    batch_results_file = args['output_file']
+    tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.DEBUG)
+    # set the input data size
+    INPUT_SIZE = args['INPUT_SIZE']
+    matrix_size = args['matrix_size']
+
+    train_dataset = Dataset(args['train_dataset_filename'])
+    test_dataset = Dataset(args['test_dataset_filename'])
+
+    exception_counter = 0
+    iteration_num = 0
+
+    parameter_names = ["learning_rate", "training_iters", "batch_size", "l2_regularization_penalty", "dropout_keep_prob",
```

Comment:
Is this defining the model?

If so, make it its own file with a sensible versioning schema.

Reply:
I have put them in a  new file: parameterset.py